storages/python/python-storage-plugins/bson.py

```diff
@@ -0,0 +1,170 @@
+                '5c65' : '1b', # Escape: '\\e' -> '\e' = hex 1b
+                '5c5c' : '5c', # Backslash: '\\\\' -> '\\' = hex 5c
+                }
+            entries = {}
+            for uuid in self.queue():
+                d = str(self.d[uuid])
+                sp = d.split("bytearray(b'")
+                d = sp.pop(0)
+                # Convert binary strings in sp to hex strings
+                for s in sp:
+                    n = s.find("')")
+                    raws = "r'" + s[:n] + "'"
+                    # NOTE 02.12.2021:
+                    # DLite corrupts binary strings in Python, but not
+                    # raw strings, so s[:n].encode('utf-8').hex()
+                    # does not work, but 'hexs' below does
+                    hexs = raws.encode('utf-8').hex()
+                    # hexs must be '7227' + h + '27', where h is a
+                    # hexadecimal string of length divisible by two,
+                    # since each binary value maps to a two-digit
+                    # hexadecimal number.
+                    # Each pair of hexadecimal digits HH in h are
+                    # on one of the following forms:
+                    # > '5cXX' = an escape sequence in esc_seq,
+                    #            e.g. '5c6e' = b'\n'
+                    # > 'XXXX' = two binary characters,
+                    #            e.g. '3130' = b'10'
+                    # > '5c78' = the first four of eight digits
+                    #            representing one binary character
+                    #            on the form '5c78XXXX',
+                    #            e.g. '5c783130' = b'\x10'
+                    hexs = hexs.lstrip('7227') # Remove "r'"
+                    hexs = hexs.rstrip('27') # Remove "'"
+                    d = d + "'"
+                    pos = 0
+                    while pos < len(hexs):
+                        wrd = hexs[pos:(pos + 4)]
+                        pos = pos + 4
+                        if wrd[0:2] == '5c':
+                            if wrd[2:4] == '78':
+                                wrd = hexs[pos:(pos + 4)]
+                                pos = pos + 4
+                                d = d + unhexlify(wrd).decode('utf-8')
+                            else:
+                                d = d + esc_seq[wrd]
+                        else:
+                            d = d + wrd
+                    d = d + "'" + s[(n + 2):]
+                entries[uuid] = d
+            with open(self.uri, mode) as f:
+                f.write(pybson.encode(entries))
+
+    def load(self, uuid):
+        """Load `uuid` from current storage and return it
+        as a new instance.
+        """
+        for key, value in self.d.items():
+            if key == uuid:
+                if type(value) == str:
+                    value = eval(value) # Translate to dict
+                # Create instance and return it
```

Comment:
Isn't it possible to just return `instance_from_dict(value)` at this point?

Reply:
For metadata, yes, For actual data, it turned out that I had to handle the hexadecimal input explicitly.

Reply:
Ahh, I see. I think the right solution would be to fix `instance_from_dict()` rather than you having to handle all the edge cases. Please add an issue.